### ISSUE TITLE
Cascading filter works only for the first value available in sub filter

### ISSUE THREAD

**Issue by PDumasBAR** [visualize-admin/visualization-tool#393]
**Describe the bug**
Dataset : The Stapfer-Enquête: schooling in Switzerland 1799 (RDF)
Chart type : Bar chart, with these options : 
![image](https://user-images.githubusercontent.com/35593772/154956608-cf4b9259-b6e9-4b46-9dbc-56ddcea7994c.png)

First and second filters work fine, with the available combination : Kanton = Argau, Bezirk = Aarau (Argau) 
1. ![image](https://user-images.githubusercontent.com/35593772/154956914-3fa6990a-ff66-42c1-a110-4b034ebc30cc.png)
2. ![image](https://user-images.githubusercontent.com/35593772/154957129-c3af6bc7-194f-4b23-89c9-d9ebaf2f66d2.png)

Filters are reset when I try to change the first filter value to Bern
3. ![image](https://user-images.githubusercontent.com/35593772/154957755-d78b4513-1f28-49f2-ba00-993be5d9fa0c.png)

Same behavior if I change first the Kanton to Bern and then add the second filter.
1. ![image](https://user-images.githubusercontent.com/35593772/154958836-e9377b09-f3ef-4d3e-a79d-8b615e446f2a.png)
2. ![image](https://user-images.githubusercontent.com/35593772/154960838-8295a8b1-c39f-4e34-8a42-451417051fb4.png)
3. ![image](https://user-images.githubusercontent.com/35593772/154960449-5ab11d0b-7d5c-4d2b-ba02-97019437707e.png)

**Expected behavior**
I already tested this functionality, maybe with another dataset, but it worked.
When I change the value in the first filter, Visualize update the second filter with a valid value.
Now, it seems Visualize tries to apply the first available value or keep the current value in the second filter when I update the value in the first filter, resulting in no available observations for the combined value.

**Environment :**
 - Visualize INT [Visualize int](https://int.visualize.admin.ch/) v3.4.1 (https://github.com/visualize-admin/visualization-tool/commit/040ed03a0445f0cc90e06d87d33336a3f31ee014)
 - Edge Version 98.0.1108.43 (Official build) (64-bit)
 - Firefox 78.12.0esr (64-bit)

**Additional context**
Tested with the dataset "Bathing water quality", same problem. The cascading filter is no more usable.

**Comment by ptbrowne** [comment]
Can you confirm it works better in test ? https://test.visualize.admin.ch/en/browse/dataset/https%3A%2F%2Fculture.ld.admin.ch%2Fsfa%2Fstapfer-enquete-1799%2F4%2F?previous=%7B%22includeDrafts%22%3Atrue%2C%22order%22%3A%22SCORE%22%2C%22search%22%3A%22stapfer%22%7D

**Comment by PDumasBAR** [comment]
Yes it's better. The filters are no more reseted when changing the value in the first filter. The second filter value is set to No Filter, 
![image](https://user-images.githubusercontent.com/35593772/154987685-4dc18a7b-3a58-4216-a036-1503980f2548.png)
but in the graph the old values are present : 
![image](https://user-images.githubusercontent.com/35593772/154987828-a7616741-600c-44dc-8fc6-289dccfb306e.png)
"No filter" for the second filter value is ok, but the chart should display all the values for the first filter.

Same thing when changing the default value in the first filter then adding a second filter.

I think, last week (or 10 days ago) the cascading filter worked very well and always displayed something. I've made a lot of tests and I can't remember a special case where it does not work.

**Comment by ptbrowne** [comment]
Thanks Pierre for your feedback, I'll take a look.

> I think, last week (or 10 days ago) the cascading filter worked very well and always displayed something. I've made a lot of tests and I can't remember a special case where it does not work.

There were cases where it did not work, see #327 and #328.
